src/Features/CSharp/Portable/MoveToResx/CSharpMoveToResxService.cs

```diff
@@ -0,0 +1,459 @@
+        {
+            return true;
+        }
+
+        // Diagnostic and telemetry
+        if (invoked.Contains("Diagnostic") || invoked.Contains("Telemetry") ||
+            invoked.Contains("Counter") || invoked.Contains("Meter") ||
+            invoked.Contains("EventSource") || invoked.Contains("Activity"))
+        {
+            return true;
+        }
+
+        // Search and query operations (technical)
+        if (invoked.Contains("ExecuteSearch") || invoked.Contains("ExecuteQuery") ||
+            invoked.Contains("Search") && (invoked.Contains("Database") || invoked.Contains("Index") || invoked.Contains("Engine")))
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    private static bool IsNonUserFacingParameterContext(ArgumentSyntax arg, InvocationExpressionSyntax invocation)
+    {
+        // Try to get parameter name from argument syntax
+        var argumentList = invocation.ArgumentList;
+        if (argumentList != null)
+        {
+            var argumentIndex = argumentList.Arguments.IndexOf(arg);
+            if (argumentIndex >= 0)
+            {
+                // Check if this is a named argument
+                if (arg.NameColon != null)
+                {
+                    var paramName = arg.NameColon.Name.Identifier.ValueText.ToLowerInvariant();
+                    return IsNonUserFacingParameterName(paramName);
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private static bool IsNonUserFacingParameterName(string paramName)
+    {
+        return paramName switch
+        {
+            "key" or "name" or "id" or "identifier" or "code" or "type" or
+            "path" or "filename" or "filepath" or "directory" or "folder" or
+            "url" or "uri" or "endpoint" or "connectionstring" or
+            "sql" or "query" or "command" or "script" or
+            "pattern" or "regex" or "format" or "template" or
+            "namespace" or "assembly" or "typename" or "methodname" or
+            "propertyname" or "fieldname" or "eventname" or
+            "category" or "source" or "level" or "severity" or
+            "configuration" or "setting" or "option" or "parameter" or
+            "value" or "config" or "env" or "environment" => true,
+            _ => false
+        };
+    }
+
+    private static bool IsInAttributeContext(LiteralExpressionSyntax stringLiteral)
+    {
+        var current = stringLiteral.Parent;
+        while (current != null)
+        {
+            if (current is AttributeSyntax)
+                return true;
+            current = current.Parent;
+        }
+        return false;
+    }
+
+    private static bool IsInPreprocessorContext(LiteralExpressionSyntax stringLiteral)
+    {
+        // Check if we're in a preprocessor directive
+        var token = stringLiteral.GetFirstToken();
+        return token.HasLeadingTrivia && token.LeadingTrivia.Any(t => t.IsKind(SyntaxKind.PreprocessingMessageTrivia));
```

Comment:
this is not the right way to check if in a pp.  just walk upwards and see if our root note is a PPDirectiveSyntax